### ISSUE TITLE
Tolerate version line errors as undef?

### ISSUE THREAD

**Issue by miyagawa** [charsbar/Parse-PMFile#8]
HTTP-Server-Simple's lib/HTTP/Server/Simple/CGI.pm as a VERSION line that can't be evaluated by its own (bug is filed: https://rt.cpan.org/Ticket/Display.html?id=38345)

Parse::PMFile's `parse` method will return an error:

```
perl -MXXX -MParse::PMFile -e 'WWW [Parse::PMFile->new->parse("lib/HTTP/Server/Simple/CGI.pm")]'    
---
- {}
- HTTP::Server::Simple::CGI:
    infile: lib/HTTP/Server/Simple/CGI.pm
    parse_version: $VERSION = $HTTP::Server::Simple::VERSION;
...
  at -e line 1
```

It looks like PAUSE indexes them however, as version `undef`. Is there any logic in PAUSE that catches these errors to round to undef, or should Parse::PMFile have an option like that? I'm sure I can catch them in my tool, but just wondering how PAUSE handles this and if there's any incompatibilities.

**Comment by charsbar** [comment]
See https://github.com/andk/pause/blob/master/lib/PAUSE/package.pm#L335 and the following block. The parser itself returns a version error, but the package is indexed with "undef" (L339, L348).

Hmm. It seems to be better that Parse::PMFile also returns "undef" version for the package, plus an error hash if required. Will also look into Parse::LocalDistribution, WorePAN and CPANTS stuff, and fix necessary stuff maybe this weekend or a bit later. Thanks.

**Comment by charsbar** [comment]
Done and released Parse::PMFile 0.35 et al. Thanks!
